app/assets/src/components/views/discovery/DiscoverySidebar/ProjectDescription.jsx

```diff
@@ -3,26 +3,81 @@ import cx from "classnames";
 import cs from "./project_description.scss";
 
 import PropTypes from "prop-types";
-import { Accordion } from "~/components/layout";
+import Textarea from "~ui/controls/Textarea";
+import MetadataSection from "~/components/common/DetailsSidebar/SampleDetailsMode/MetadataSection";
 
+import { saveProjectDescription } from "~/api";
 import { MAX_DESCRIPTION_LENGTH } from "~/components/views/projects/constants";
 
 class ProjectDescription extends React.Component {
   constructor(props) {
     super(props);
 
     this.toggleDisplayDescription = this.toggleDisplayDescription.bind(this);
+    this.toggleEditing = this.toggleEditing.bind(this);
 
     this.state = {
       description: props.project.description,
+      lastValidDescription: props.project.description,
       showLess: true,
+      editing: false,
+      changed: false,
+      savePending: false,
     };
   }
 
   toggleDisplayDescription() {
     this.setState(prevState => ({ showLess: !prevState.showLess }));
   }
 
+  toggleEditing() {
+    this.setState(prevState => ({ editing: !prevState.editing }));
+  }
+
+  handleDescriptionChange = value => {
+    this.setState({
+      description: value,
+      changed: true,
+    });
+  };
+
+  handleDescriptionSave = async () => {
+    if (this.state.changed) {
+      this.setState(
+        {
+          changed: false,
+        },
+        () => {
+          this._save(this.props.project.id, this.state.description);
+        }
+      );
+    }
+  };
+
+  _save = async (projectId, value) => {
```

Comment:
I don't think we follow the underscore convention for private JS methods, though I like it. I'd like to hear others's thoughts.

Reply:
It's used in `SampleDetailsMode.jsx` to save metadata details as well, which is the style I was following!